libs/sql-ddl/src/mysql.rs

```diff
@@ -0,0 +1,213 @@
+use std::{borrow::Cow, fmt::Display};
+
+#[derive(Debug, Default)]
+pub struct AlterTable<'a> {
+    pub table_name: Cow<'a, str>,
+    pub changes: Vec<AlterTableClause<'a>>,
+}
+
+impl Display for AlterTable<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "ALTER TABLE `{}`", self.table_name)?;
+
+        if self.changes.len() == 1 {
+            write!(f, " {}", self.changes[0])?;
+
+            return Ok(());
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Debug)]
+pub enum AlterTableClause<'a> {
+    AddForeignKey(ForeignKey<'a>),
+    DropColumn {
+        column_name: Cow<'a, str>,
+    },
+    DropForeignKey {
+        constraint_name: Cow<'a, str>,
+    },
+    DropPrimaryKey,
+    RenameIndex {
+        previous_name: Cow<'a, str>,
+        next_name: Cow<'a, str>,
+    },
+    RenameTo {
+        next_name: Cow<'a, str>,
+    },
+}
+
+impl Display for AlterTableClause<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            AlterTableClause::RenameTo { next_name } => write!(f, "RENAME TO {}", next_name),
+            AlterTableClause::RenameIndex {
+                previous_name,
+                next_name,
+            } => write!(f, "RENAME INDEX `{}` TO `{}`", previous_name, next_name),
+            AlterTableClause::DropColumn { column_name } => write!(f, "DROP COLUMN `{}`", column_name),
+            AlterTableClause::DropForeignKey { constraint_name } => write!(f, "DROP FOREIGN KEY `{}`", constraint_name),
+            AlterTableClause::DropPrimaryKey => f.write_str("DROP PRIMARY KEY"),
+            AlterTableClause::AddForeignKey(fk) => write!(f, "ADD {}", fk),
+        }
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct ForeignKey<'a> {
+    pub constraint_name: Option<Cow<'a, str>>,
+    pub constrained_columns: Vec<Cow<'a, str>>,
+    pub referenced_table: Cow<'a, str>,
+    pub referenced_columns: Vec<Cow<'a, str>>,
+    pub on_delete: Option<ForeignKeyAction>,
+    pub on_update: Option<ForeignKeyAction>,
+}
+
+impl<'a> Display for ForeignKey<'a> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if let Some(constraint_name) = &self.constraint_name {
+            write!(f, "CONSTRAINT `{constraint_name}` ", constraint_name = constraint_name,)?;
+        }
+
+        f.write_str("FOREIGN KEY (")?;
+
+        let mut constrained_columns = self.constrained_columns.iter().peekable();
+
+        while let Some(column) = constrained_columns.next() {
+            write!(f, "`{}`", column)?;
+
+            if constrained_columns.peek().is_some() {
+                f.write_str(", ")?;
+            }
+        }
+
```

Comment:
Is there a reason why you're not using your join iterator here?

Reply:
Did just that, good suggestion, thanks!